### PR TITLE
Fix slug validation and generation for Attribute create

### PR DESCRIPTION
**Pull request by jakubkuc**
I want to merge this change because it fixes a problem where creating Attribute with two similar values returned ValidationError when it shouldn't. 

Example valid input that was raising error: 
```{
  "input": {
    "availableInGrid": true,
    "entityType": null,
    "filterableInDashboard": true,
    "filterableInStorefront": true,
    "inputType": "DROPDOWN",
    "name": "test",
    "slug": "test",
    "storefrontSearchPosition": null,
    "type": "PRODUCT_TYPE",
    "valueRequired": true,
    "visibleInStorefront": true,
    "values": [
      {
        "name": "1.5"
      },
      {
        "name": "15"
      }
    ]
  }
}
```

Acceptance criteria: 
- Mutation with given input works properly and creates Attribute with given values.
- Adding values with similar names (the ones that would have the same result when they are slugified) would result in generating a unique slug for all the values. 
- Names that are different only by upper/lower case or spaces are not allowed.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by maarcingebala — COMMENTED** [review]
Can we also add a test that ensures that the case for which the bug was reported is working correctly?

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/attribute/tests/mutations/test_attribute_update.py
Comment (IKarbowiak) [inline saleor/graphql/attribute/tests/mutations/test_attribute_update.py:1]: Could you also add a test for creating attribute values with names that give the same slug with slugify?

Comment (jakubkuc) [reply saleor/graphql/attribute/tests/mutations/test_attribute_update.py:1]: added 👍🏻

File: saleor/graphql/attribute/tests/mutations/test_attribute_create.py [hunk 1]
```diff
@@ -1255,3 +1255,43 @@ def test_create_attribute_and_attribute_values_errors(
     assert errors[0]["field"] == "values"
     assert errors[0]["message"] == error_msg
     assert errors[0]["code"] == error_code.name
+
+
+def test_create_attribute_similar_names(
```
Comment (IKarbowiak) [inline saleor/graphql/attribute/tests/mutations/test_attribute_create.py:1260]: Could we also add a similar one for `update_attribute` mutation, when the value with `15` already exists and we are adding a new `1.5` value?

**Review by szczecha — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

File: saleor/graphql/attribute/mutations/mixins.py [hunk 2]
```diff
@@ -115,10 +121,10 @@ def validate_swatch_attr_value(cls, value_data: dict):
     @classmethod
     def check_values_are_unique(cls, values_input: dict, attribute: models.Attribute):
         # Check values uniqueness in case of creating new attribute.
-        existing_values = attribute.values.values_list("slug", flat=True)
+        existing_names = attribute.values.values_list("slug", flat=True)
```
Comment (IKarbowiak) [inline saleor/graphql/attribute/mutations/mixins.py:124]:
```suggestion
        existing_names = attribute.values.values_list("name", flat=True)
```

**Review by IKarbowiak — APPROVED** [review]

**Comment by slawekmarszalek** [comment]
Reported in: https://saleorcommerce.slack.com/archives/C038XPD8HLK/p1678892312013259

**Pull request merged**